src/components/Navigator/QuickNavigationPreview.vue

```diff
@@ -0,0 +1,136 @@
+-->
+
+<template>
+  <div class="preview">
+    <!-- success, show a preview -->
+    <DocumentationTopic
+      v-if="state === STATE.success"
+      v-bind="topicProps"
+      enableMinimized
+    />
+    <!-- only show a loader in the UI if things are loading slowly -->
+    <!-- otherwise, show nothing for the default loading state -->
+    <!-- TODO: extract a more generic loader component in the future -->
+    <div v-else-if="state === STATE.loadingSlowly" class="loading">
+      <div
+        v-for="style in LOADER_ROW_STYLES"
+        class="loading-row"
+        :key="style['--index']"
+        :style="style"
+      />
+    </div>
+    <!-- error, unavailable message -->
+    <div v-else-if="state === STATE.error" class="unavailable">
+      <p>Preview unavailable</p>
+    </div>
+  </div>
+</template>
+
+<script>
+import DocumentationTopic from 'docc-render/components/DocumentationTopic.vue';
+
+const { extractProps } = DocumentationTopic.methods;
+
+const HERO_KIND = 'hero';
+const STATE = {
+  error: 'error',
+  loading: 'loading',
+  loadingSlowly: 'loadingSlowly',
+  success: 'success',
+};
+
+export default {
+  name: 'QuickNavigationPreview',
+  components: { DocumentationTopic },
+  constants: { PreviewState: STATE },
+  props: {
+    json: {
+      type: Object,
+      required: false,
+    },
+    state: {
+      type: String,
+      required: true,
+      validator: v => Object.hasOwnProperty.call(STATE, v),
+    },
+  },
+  computed: {
+    LOADER_ROW_STYLES: () => ([
+      { '--index': 0, width: '30%' },
+      { '--index': 1, width: '80%' },
+      { '--index': 2, width: '50%' },
+    ]),
+    STATE: () => STATE,
+    topicProps: ({ json }) => {
+      const props = extractProps(json);
+
+      // massage the render JSON for both /documentation/* and /tutorials/*
+      // pages into props that can be safely rendered using a minimized
+      // `DocumentationTopic` component
+      //
+      // for /tutorials/* pages, this means extracting the first `sections`
+      // hero item and using its content as the `abstract`
+      const { sections = [] } = json;
+      let { abstract } = props;
+      const hero = sections.find(({ kind }) => kind === HERO_KIND);
+      if (!abstract && hero) {
+        abstract = hero.content;
+      }
+
+      return {
+        ...props,
+        abstract,
+      };
+    },
+  },
+};
+</script>
+
+<style scoped lang="scss">
+@import 'docc-render/styles/_core.scss';
+
+.preview {
+  @include breakpoint(small) {
+    display: none;
+  }
+}
+
+.unavailable {
+  align-items: center;
+  display: flex;
+  height: 100%;
+  justify-content: center;
+}
+
+.loading {
+  padding: 20px;
+
+  &-row {
+    animation: pulse 2.5s ease;
+    animation-delay: calc(1s + 0.3s * var(--index));
+    animation-fill-mode: forwards;
+    animation-iteration-count: infinite;
+    background-color: var(--color-fill-gray-tertiary);
+    border-radius: 4px;
+    height: 12px;
```

Comment:
In the future, as a refactor it would be nice to extract the styles of each loader row and create a component that we could reuse for the loader at  `LoadingNavigatorItem` .

Reply:
Agreed. I looked into doing this but thought it would take too much effort/time since the design of the rows differs slightly here compared to the navigator.

We should try to make a more general component in the future that both places can use for sure though. I mostly copied your styling for the navigator loader here 😁